src/tilemaps/components/TileToWorldX.js

```diff
@@ -33,7 +34,19 @@ var TileToWorldX = function (tileX, camera, layer)
         tileWidth *= tilemapLayer.scaleX;
     }
 
-    return layerWorldX + tileX * tileWidth;
+    if (orientation === 'orthogonal')
+    {
+        return layerWorldX + tileX * tileWidth;
+    }
+    else if (orientation === 'isometric' || orientation === 'staggered' || orientation === 'hexagonal')
```

Comment:
I'd suggest defining constants for tilemap layer orientations instead of relying on specific string values here. Take a look at how constants are defined elsewhere in the framework.

Reply:
It might actually be better to have entirely separate classes, or at least tilemap _components_, to avoid this branching for the sake of performance.

By component, I mean an object or system of some kind that can represent a tilemap orientation and handle their differences, such as coordinate space conversion as demonstrated here.

This means tilemaps can be _composed_ with an orientation, instead of determining them at runtime. Changing an existing tilemap's orientation would then be a case of swapping out the underlying orientation implementation.

Reply:
About the constants, it's certainly better than comparing strings.
About splitting the implementation differently, yeah, you're probably right. I don't think it matters _that_ much for performance, (there are tons of other branchings in a lot of often used components ) but it's - really - not clean.